### PR TITLE
Adds support for GZIP compressed CSV files

### PR DESCRIPTION
**Pull request by cuducos**
# Overview

As described in #94, this PR adds support for GZIP compressed CSV files in `csv.FromFile`. Also, it adds an example in the `examples/compressed/` directory — I used that to manually test it since I saw no `testdata/` directory or any other test using file fixtures.

---

Please preserve this line to notify @danielfireman (lead of this repository)

---
This change is [Reviewable](https://reviewable.io/reviews/frictionlessdata/tableschema-go/96)

### CODE REVIEW

File: csv/table.go [hunk 1]
```diff
@@ -175,13 +177,22 @@ type CreationOpts func(t *Table) error
...
 			return nil, err
 		}
+
+		ext := strings.ToLower(filepath.Ext(path))
+		if ext == ".gz" || ext == ".gzip" {
+			g, err := gzip.NewReader(f)
+			if err != nil {
+				return nil, err
+			}
+			return g, nil
```
Comment (danielfireman) [inline csv/table.go:194]:
```suggestion
			return gzip.NewReader(f)
```

**Review by danielfireman — APPROVED** [review]
Super cool! Thanks a lot for this!

Could you please take a look at my small suggestion bellow?

File: csv/table.go (hunk 1 above)
Comment (cuducos) [reply csv/table.go]: Wow! Sure : )

**Review by danielfireman — APPROVED** [review]

**Comment by danielfireman** [comment]
Tanks a lot, @cuducos !

**Pull request merged**